crates/sel4-capdl-initializer/core/src/lib.rs

```diff
@@ -568,7 +696,8 @@ impl<'a, N: ObjectName, D: Content, M: GetEmbeddedFrame, B: BorrowMut<[PerObject
             .filter_objects_with::<&object::PageTable>(|obj| obj.is_root)
         {
             let vspace = self.orig_cap::<cap_type::VSpace>(obj_id);
-            self.init_vspace(vspace, 0, 0, obj)?;
+            let root_level = obj.level.unwrap().into();
```

Comment:
In the case of RISC-V, `obj.level` is always `None`. 

Changing this line to the following would handle that case:

```
let root_level = obj.level.unwrap_or(0).into();
```